### PR TITLE
New: Added enabledTypes property. Backwards compat for disabledTypes

### PR DESCRIPTION
**Pull request by JustinHoldstock**
Adding `enabledTypes` to the annotation configuration, as well as a default enabled annotation types(`DEFAULT_ENABLED`), if none provided, to `ANNOTATORS`. This allows us to specify all of the *allowed* kinds of annotators, and the ones that are enabled by default (if no config provided).

IE)
```
// for document annotator, results in only 'point' annotations, as timestamp are not allowed
enabledTypes: ['point', 'timestamp']
```

### CODE REVIEW

File: src/lib/annotations/BoxAnnotations.js [hunk 1]
```diff
@@ -4,18 +4,27 @@ import DrawingModeController from './drawing/DrawingModeController';
...
+        TYPE: [TYPES.point, TYPES.highlight, TYPES.highlight_comment, TYPES.draw],
+        DEFAULT_ENABLED: [TYPES.point, TYPES.highlight, TYPES.highlight_comment]
     },
     {
         NAME: 'Image',
         CONSTRUCTOR: ImageAnnotator,
         VIEWER: ['Image', 'MultiImage'],
-        TYPE: [TYPES.point]
+        TYPE: [TYPES.point],
+        DEFAULT_ENABLED: [TYPES.point]
```
Comment (jeremypress) [inline src/lib/annotations/BoxAnnotations.js]: I agree with using `DEFAULT_TYPES`

File: src/lib/annotations/BoxAnnotations.js [hunk 2]
```diff
@@ -104,12 +113,19 @@ class BoxAnnotations {
 
         modifiedAnnotator = Object.assign({}, annotator);
 
-        // Filter out disabled annotation types
-        if (Array.isArray(viewerConfig.disabledTypes)) {
-            modifiedAnnotator.TYPE = modifiedAnnotator.TYPE.filter((type) => {
-                return !viewerConfig.disabledTypes.includes(type);
-            });
-        }
+        const enabledTypes = viewerConfig.enabledTypes || [...modifiedAnnotator.DEFAULT_ENABLED];
```
Comment (jeremypress) [inline src/lib/annotations/BoxAnnotations.js]: Why do we need the spread operator and then an array? Isn't `modifiedAnnotator.DEFAULT_ENABLED` already an array?

Comment (JustinHoldstock) [reply src/lib/annotations/BoxAnnotations.js]: I want to make a copy of the array, not modify the original

File: src/lib/annotations/BoxAnnotations.js (hunk 1 above)
Comment (JustinHoldstock) [reply src/lib/annotations/BoxAnnotations.js]: done

File: src/lib/annotations/README.md [hunk 3]
```diff
@@ -139,15 +139,16 @@ preview.show(..., {
         VIEWER_NAME: {
             annotations: {
                 enabled: Boolean, // Enables/disables if set. Respects "showAnnotations" if empty
-                disabledTypes: String[] // List of annotation types to disable
+                enabledTypes: String[] | null // List of annotation types to enable for this viewer. If empty, will respect default types for that annotator.
+                disabledTypes: String[] | null // DEPRECATED. List of annotation types to disable.
```
Comment (pramodsum) [inline src/lib/annotations/README.md]: let's not add this line so people won't use disabledTypes from here on out

**Pull request merged**